docs/compatible_microcontrollers.md

```diff
@@ -62,6 +62,52 @@ You can also use any ARM chip with USB that [ChibiOS](https://www.chibios.org) s
+| serial driver    | :heavy_check_mark: using `SIO` + `PIO` driver |
+| UART driver      | Support planed                                |
+
+#### GPIO pin nomenclature
+
+To address individual pins QMK uses the `GPx` abbreviation, the `x` stands for the GPIO number of the pin. This can usually be found on the official pinout diagramm of your board. Please note that these GPIO numbers not necessarily match the number you see printed on the development board. For instance the [Raspberry Pi Pico](https://datasheets.raspberrypi.com/pico/Pico-R3-A4-Pinout.pdf) uses numbers from 1 to 40 for their pins, but these are not the internal GPIO numbers. So if you want to use the pin 11 of the Pico for your keyboard, you would reefer to it as `GP8` in the config files.
+
+!> The GPIO pins of the RP2040 are not 5V tolerant!
+
+#### Double-tap reset boot-loader entry
+
+The double-tap reset mechanism is an alternate way in QMK to enter the embedded mass storage UF2 boot-loader of the RP2040. It works by a fast double-tap of the reset pin on start up, which is similar to the behavior of AVR Pro Micros. This feature is not activated by default and has to be configured. To activate it add the following options to your keyboards `config.h` file:
+
+```c
+#define RP2040_BOOTLOADER_DOUBLE_TAP_RESET // Activates the double-tap behavior
+#define RP2040_BOOTLOADER_DOUBLE_TAP_RESET_TIMEOUT 200U // Timeout window in ms in which the double tap can occur.
+#define RP2040_BOOTLOADER_DOUBLE_TAP_RESET_LED_MASK 0U // Specify a optional status led which blinks when entering the bootloader
+```
+
+#### Split keyboard support
+
+Split keyboards are fully supported using the [serial driver](serial_driver.md) in both Full-duplex and Half-duplex configurations. For this to work two driver subsystems are supported by the RP2040, the hardware UART based `SIO` and the Programmable IO based `PIO` driver.
+
+| Feature                       | [SIO Driver](serial_driver.md#the-sio-driver) | [PIO Driver](serial_driver.md#the-pio-driver) |
+| ----------------------------- | --------------------------------------------- | --------------------------------------------- |
+| Half-Duplex operation         |                                               | :heavy_check_mark:                            |
+| Full-Duplex operation         | :heavy_check_mark:                            | :heavy_check_mark:                            |
+| `TX` and `RX` pin swapping    |                                               | :heavy_check_mark:                            |
+| Any GPIO as `TX` and `RX` pin | Only UART capable pins                        | :heavy_check_mark:                            |
+| Simple configuration          |                                               | :heavy_check_mark:                            |
+
+The `PIO` driver is much more flexible then the `SIO` driver, the only "downside" is the usage of `PIO` resources which in turn are not available for avanced user programs.
```

Comment:
Does the driver choice and full vs. half duplex have any human-noticeable impact on performance, scan rate, etc.?

Reply:
> Does the driver choice and full vs. half duplex have any human-noticeable impact on performance, scan rate, etc.?

Pretty sure "no".  However, the issue is with hardware implementation (eg, pcb hookup) requires different configuration based on the type. 

Full duplex would require wiring TX <-> RX, which many split boards are NOT designed for.  and I don't know about the 2040, but for stm32 half duplex requires a pullup (and I'm guessing so does the 2040), which most boards are not configured with (but many have the wiring for, due to overlap with i2c support). 

Though, I'm not certain about this.

Reply:
> Does the driver choice and full vs. half duplex have any human-noticeable impact on performance, scan rate, etc.?

Basically what drashna already said, the scan rate is well above 1KHz with the default baudrate, even with RGB and OLEDs enabled. The deciding factor should be the hardware layout, the only advantage I see for the SIO driver is for applications that need both PIO peripherals. 

>  and I don't know about the 2040, but for stm32 half duplex requires a pullup (and I'm guessing so does the 2040), which most boards are not configured with (but many have the wiring for, due to overlap with i2c support).

The RP2040 does indeed provide a pull-up that is enough for Half-duplex operation, so even for retrofitted keyboard like a corne or kyria doesn't need an external pull-up.